### DISCUSSION TITLE
Does Huma currently support multiple accepted content types for raw bodies?

### DISCUSSION THREAD

**Discussion by sm3142** [danielgtaylor/huma#682] [Q&A]
OpenAPI 3 allows to declare multiple accepted content types for a given endpoint: https://swagger.io/docs/specification/v3_0/describing-request-body/describing-request-body.

I have a use case that would benefit from this because the raw body can either be encoded as JSON or in binary format. 

E.g. something along the lines of:
```
type Request struct {
	RawBody []byte `contentType:"application/octet-stream,application/json`
}
```
(using a comma-separated list in real life is probably a bad idea, it just used it here for illustrative purposes)

The handler function naturally has to treat the data according to a given request's actual content type. As per usual when using `RawBody` interpretation of the the data is entirely up the handler. 

Is something like this already supported, i.e. am I just too stupid to read the documentation? If not, would it make sense for me to investigate how such a feature could be added - or is that a fool's errand?

PS: I guess you can achieve something similar with multi-part forms where each part is optional, but semantically I find this somewhat janky.

**Comment by danielgtaylor** [comment]
Since the raw body isn't parsed, I believe you can leave off the `contentType` field tag and manually supply a schema with your `huma.Operation`, then it can include any number of content types you want.
